src/h5web/vis-packs/core/hooks.ts

```diff
@@ -13,24 +13,19 @@ import {
 } from './utils';
 import AxisSystemContext from './shared/AxisSystemContext';
 import type { AxisScale } from './models';
-import type { HDF5Shape, HDF5Type } from '../../providers/hdf5-models';
 import { ProviderContext } from '../../providers/context';
 import type { Dataset, Value } from '../../providers/models';
 import { isAxis } from '../../dimension-mapper/utils';
 
-export function useDatasetValue<
-  S extends HDF5Shape,
-  T extends HDF5Type,
-  D extends Dataset<S, T> | undefined
->(
+export function useDatasetValue<D extends Dataset | undefined>(
   dataset: D,
   dimMapping?: DimensionMapping
-): D extends Dataset<infer S, infer T> ? Value<S, T> : undefined;
+): D extends Dataset ? Value<D> : undefined;
 
 export function useDatasetValue(
   dataset: Dataset | undefined,
   dimMapping?: DimensionMapping
-) {
+): unknown {
```

Comment:
Adding explicit return type for clarity and performance reasons.

Reply:
Could you elaborate on this ? I don't get the reasons :smile:

Reply:
- Performance (granted, it's probably not a big deal in this case... 😄): https://github.com/microsoft/TypeScript/wiki/Performance#using-type-annotations
- The benefit I mention in my other comment also applies here: I'm now clearly identifying the typing of the function's internal declaration (as opposed to the typing of the function's external usage provided by the override). Since the two typings are different, explicitness seems worthwhile.
- The return type is short and leaving it out doesn't improve the formatting of the code (the function signature still gets wrapped over multiple lines), so there's no downside to having it, really.

That being said, I could use `HDF5Value` instead of `unknown`, I forgot about it!